ecosystem/indexer/src/models/token.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{models::events::Event, schema::tokens};
+use aptos_rest_client::types;
+use std::{collections::HashMap, fmt, fmt::Formatter, str::FromStr};
+
+use serde::{Deserialize, Deserializer, Serialize};
+
+#[derive(Associations, Debug, Identifiable, Insertable, Queryable, Serialize)]
+#[diesel(table_name = "tokens")]
+#[primary_key(token_id)]
+pub struct Token {
+    pub token_id: String,
+    pub creator: String,
+    pub collection: String,
+    pub name: String,
+    pub description: String,
+    pub max_amount: Option<i64>,
+    pub supply: i64,
+    pub uri: String,
+    pub minted_at: chrono::NaiveDateTime,
+
+    pub inserted_at: chrono::NaiveDateTime,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TokenId {
+    pub creator: String,
+    pub collection: String,
+    pub name: String,
+}
+
+impl TokenId {
+    pub fn from_serde_value(value: serde_json::Value) -> Self {
+        serde_json::from_value(value).unwrap()
+    }
+}
+
+impl fmt::Display for TokenId {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        write!(f, "{}::{}::{}", self.creator, self.collection, self.name)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TokenData {
+    pub collection: String,
+    pub description: String,
+    pub name: String,
+    #[serde(deserialize_with = "deserialize_option_from_string")]
+    pub maximum: MoveOption<i64>,
+    #[serde(deserialize_with = "deserialize_option_from_string")]
+    pub supply: MoveOption<i64>,
+    pub uri: String,
```

Comment:
Is this the actual image uri or the arweave uri to the json? If its the actual image uri can we make the field name `image`?

Reply:
this is the metadata uri to the json.
I will create another metadata table, which we can have a field called URL?

Reply:
Yeah, maybe make a table called `NFTMetadata` or something like that (other tokens will have different metadata schemas)

Reply:
sure